rules/framework.bzl

```diff
@@ -1204,8 +1204,9 @@ that this target depends on.
             """,
         ),
         "_child_configuration_dummy": attr.label(
-            cfg = transition_support.split_transition,
-            default = Label("@bazel_tools//tools/cpp:current_cc_toolchain"),
+            cfg = transition_support.apple_platform_split_transition,
+            providers = [cc_common.CcToolchainInfo, ApplePlatformInfo],
+            default = "@build_bazel_rules_apple//apple:default_cc_toolchain_forwarder",
```

Comment:
After I updated `transition_support.bzl` to match rules_apple I was getting a Starlark crash. This change fixes it and matches rules_apple: https://github.com/bazelbuild/rules_apple/blob/f8fa4d9a32556ece0f7eee0ebdf997ca25bdc855/apple/internal/rule_attrs.bzl#L140-L149

Reply:
I'll follow up this PR with another to update toolchains, it'll fix toolchain resolution errors on Bazel 7

Reply:
Moving this to #848